File: frontend/src/domain/questions/Question.ts
import {QuestionType} from "./QuestionType";

export interface Question {
    id: number;
    number: string;
    name: string;
    blockName: string;

    media: string;
    questionType: QuestionType;
    points: number;
    categories: string[];
    law: string;
    lawDescription: string;
    securityThemOfQuestion: string;

    questionTextPL: string;
    answerATextPL: string;
    answerBTextPL: string;
    answerCTextPL: string;

    questionTextENG: string;
    answerATextENG: string;
    answerBTextENG: string;
    answerCTextENG: string;

    questionTextDE: string;
    answerATextDE: string;
    answerBTextDE: string;
    answerCTextDE: string;

    rightAnswer: string;

    knowType: string;
    bookmarked: boolean;
}

File: backend/src/main/java/com/mokhnachev/prawaruchu/questions/domain/Question.java
package com.mokhnachev.prawaruchu.questions.domain;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@NoArgsConstructor
@Data
public class Question {
    private Long id;
    private String number;
    private String name;
    private String blockName;

    private String media;
    private QuestionType questionType;
    private int points;
    private List<Category> categories;
    private String law;
    private String lawDescription;
    private String securityThemOfQuestion;

    private String questionTextPL;
    private String answerATextPL;
    private String answerBTextPL;
    private String answerCTextPL;

    private String questionTextENG;
    private String answerATextENG;
    private String answerBTextENG;
    private String answerCTextENG;

    private String questionTextDE;
    private String answerATextDE;
    private String answerBTextDE;
    private String answerCTextDE;

    private RightAnswer rightAnswer;

    private KnowType knowType;
    private Boolean bookmarked;
}
